app/src/main/java/com/mapbox/maps/testapp/examples/OfflineActivity.kt

```diff
@@ -0,0 +1,329 @@
+    // Note this will not remove the downloaded style pack, instead, it will just mark the resources
+    // not a part of the existing style pack. The resources still exists as ambient cache.
+    if (this::offlineManager.isInitialized) {
+      offlineManager.removeStylePack(Style.OUTDOORS)
+    }
+    // Remove the existing style resources from ambient cache using cache manager.
+    val cacheManager = CacheManager(
+      MapInitOptions.getDefaultResourceOptions(this)
+    )
+    cacheManager.clearAmbientCache {}
+
+    // Reset progressbar.
+    updateStylePackDownloadProgress(0)
+    updateTileRegionDownloadProgress(0)
+  }
+
+  private fun updateStylePackDownloadProgress(progress: Int, message: String? = null) {
+    style_pack_download_progress.progress = progress
+    message?.let {
+      offlineLogsAdapter.addLog(OfflineLog.StylePackProgress(it))
+    }
+  }
+
+  private fun updateTileRegionDownloadProgress(progress: Int, message: String? = null) {
+    tile_pack_download_progress.progress = progress
+    message?.let {
+      offlineLogsAdapter.addLog(OfflineLog.TilePackProgress(it))
+    }
+  }
+
+  private fun logInfoMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Info(message))
+  }
+
+  private fun logErrorMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Error(message))
+  }
+
+  private fun logSuccessMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Success(message))
+  }
+
+  override fun onStart() {
+    super.onStart()
+    mapView?.onStart()
+  }
+
+  override fun onStop() {
+    super.onStop()
+    mapView?.onStop()
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    removeOfflineRegions()
+    mapView?.onDestroy()
+  }
+
+  private class OfflineLogsAdapter : RecyclerView.Adapter<OfflineLogsAdapter.ViewHolder>() {
+    private var isUpdating: Boolean = false
+    private val updateHandler = Handler()
+    private val logs = ArrayList<OfflineLog>()
+
+    private val updateRunnable = Runnable {
+      notifyDataSetChanged()
+      isUpdating = false
+    }
+
+    class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
+      internal var alertMessageTv: TextView = view.findViewById(R.id.alert_message)
+    }
+
+    @NonNull
+    override fun onCreateViewHolder(@NonNull parent: ViewGroup, viewType: Int): ViewHolder {
+      val view =
+        LayoutInflater.from(parent.context).inflate(R.layout.item_gesture_alert, parent, false)
+      return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(@NonNull holder: ViewHolder, position: Int) {
+      val alert = logs[position]
+      holder.alertMessageTv.text = alert.message
+      holder.alertMessageTv.setTextColor(
+        ContextCompat.getColor(holder.alertMessageTv.context, alert.color)
+      )
+    }
+
+    override fun getItemCount(): Int {
+      return logs.size
+    }
+
+    fun addLog(alert: OfflineLog) {
+      logs.add(0, alert)
```

Comment:
Can we also output the log to logcat here? It can help us filtering logs

Reply:
added